eval/src/vespa/eval/eval/value_type.cpp

```diff
@@ -388,6 +388,12 @@ ValueType::either(const ValueType &one, const ValueType &other) {
     return one;
 }
 
+ValueType
+ValueType::cell_cast(const ValueType &from, CellType to_cell_type) {
+    // XXX we want to change this and use make_type later:
+    return tensor_type(from.dimensions(), to_cell_type);
```

Comment:
I think we should do make_type here already and rather make type-resolving decide what is legal or not